pyproject.toml

```diff
@@ -0,0 +1,39 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+#build-backend = "setuptools.build_meta"
+
+
+[tool.setuptools]
+packages = ["usd_qtpy"]
+
+
+[project]
+name = "usd-qtpy"
+authors = [
+    {name = "BigRoy"},
+    {name = "Hannes"},
+]
+description = "Python Qt components for building custom USD tools."
+readme = "README.md"
+requires-python = ">=3.4"
+keywords = ["usd", "qt"]
+license = { file = "LICENSE" }
+classifiers = [
+    "Programming Language :: Python :: 3.7",
+]
+dependencies = [
+    'importlib-metadata; python_version<"3.7"', "qtpy", "usd-core"
```

Comment:
Bear with me as I have little knowledge of these pyproject files.

1. Why does this need `importlib-metadata` for lower Python versions?

---

2. I'm a bit unsure about including `usd-core` - yes it is a dependency, but it is far more likely that users might be rolling their own USD builds in production. `usd-core` for example does not include `usdviewq` and thus cannot display the USD Viewport then.

As far as I know `usdviewq` would need to build USD from source - and that would also bring in e.g. an additional dependency `PyOpenGL` to actually be able to render in the viewport.

Anyway, if we _want_ users to use it against their own USD libs (that I think might conflict with `pip install usd-core`; no idea) should we still include it as a dependency?

Reply:
1. good question, this is the template i always use for my own pypi packages, it works so i dont change it.
i think you can remove it from dependencies.
i wonder if this means to get `importlib-metadata` dependency for old python versions

2. not sure how pip would handle that, if you roll your own.
you can also add an optional dependency in the toml.

Reply:
> you can also add an optional dependency in the toml.

Interesting - I wonder if we should do that instead - maybe at least for `PyOpenGL` (which is not listed now) since that's optional and needed only when combining it with `prx.Usdviewq`. :) but maybe `usd-core` is also an option for that.`

Reply:
```toml
[project.optional-dependencies]
usd = ["usd-core"]
usdview = ["PyOpenGL"]
```

This might then suffice?

To install with `usd` core would then be `pip install usd_qtpy[usd]` right?

Or:
```
pip install usd_qtpy[usd, usdview]
```

However there's no ready to go builds I believe for `usdview` itself - it'd still need a build of your own so it'd be quite an incomplete optional dependency. But building USD view with `usdviewq` does not necessarily mean you get `PyOpenGL` so it might be good to get it listed as optional dependency here?